### PR TITLE
Pupil teacher chart

### PR DESCRIPTION
**Pull request by esoliani**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -131,6 +131,7 @@
     <script src="scripts/directives/gaugechart.js"></script>
     <script src="scripts/directives/passratechange.js"></script>
     <script src="scripts/directives/passratetime.js"></script>
+    <script src="scripts/directives/pupilteacherratiochart.js"></script>
```
Comment (uniphil) [inline app/index.html]: can we move these `scripts/directives/*` up with the other directives at line 104?

File: app/scripts/controllers/dashboard.coffee [hunk 2]
```diff
@@ -158,6 +158,7 @@ angular.module('edudashAppCtrl').controller 'DashboardCtrl', [
 
         $scope.setSchool = (item, model, showAllSchools) ->
             $scope.selectedSchool = item
+            $scope.selectedSchool['pupil-teacher-ration'] = 56
```
Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee:161]: Is this for debug until we have real data? Looking at `item` here, there is a `item.pt_ratio` available that might be ok to use.

File: app/scripts/directives/pupilteacherratiochart.coffee [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ # @name edudashApp.directive:ratePassChart
+ # @description
+ # # ratePassChart
+###
+angular.module 'edudashAppDir'
+.directive 'pupilTeacherRatioChart', [
+    '$log'
+    ($log) ->
+      restrict: 'E'
+      templateUrl: 'views/pupilteacherratiochart.html'
```
Comment (uniphil) [inline app/scripts/directives/pupilteacherratiochart.coffee:14]: If we add

``` coffeescript
    scope: {
      selectedSchool: '=selectedSchool'
    }
```

to the directive, then we can pass `selectedSchool` as an `attr` and make the dependency explicit, and make this directive a little safer.

File: app/scripts/directives/pupilteacherratiochart.coffee [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      link: (scope, element, attrs) ->
+        max = attrs.max
+        min = attrs.min
+        update = () ->
+          if scope.selectedSchool? then element.show() else element.hide()
+          ptRatio = scope.selectedSchool['pupil-teacher-ration']
+          if ptRatio <= min
+            src = 'images/passrate_student_green.png'
+            textClass = 'text-gree'
```
Comment (uniphil) [inline app/scripts/directives/pupilteacherratiochart.coffee]: I think this should be `text-green`

File: app/scripts/directives/pupilteacherratiochart.coffee [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            src = 'images/passrate_student_yellow.png';
+            textClass = 'text-yellow'
+          else
+            src = 'images/passrate_student_red.png';
+            textClass = 'text-red'
+
+          element.find('.imageStudent').attr('src', src)
+          element.find('.percentNumber').attr('class', "percentNumber widgetnumber #{textClass}")
+
+        scope.$watch 'selectedSchool', (newValue, oldValue) ->
```
Comment (uniphil) [inline app/scripts/directives/pupilteacherratiochart.coffee:39]: cool :+1:

File: app/views/pupilteacherratiochart.html [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<div class="widget first">
+    <p class="widgettitle" translate="chart.pupil-teacher-ration"></p>
+    <div class="row">
+        <div class="col-xs-2"><p class="percentNumber widgetnumber text-yellow">{{ selectedSchool['pupil-teacher-ration'] }}%</p>
```
Comment (uniphil) [inline app/views/pupilteacherratiochart.html]: I think it's not a percent, so we should drop the `%`.

File: app/scripts/controllers/dashboard.coffee (hunk 2 above)
Comment (esoliani) [inline app/scripts/controllers/dashboard.coffee:161]: pt_ratio was always null. I hardcoded to test it. we need to change the data anyway

Comment (uniphil) [inline app/scripts/controllers/dashboard.coffee:161]: sounds good

File: app/scripts/directives/pupilteacherratiochart.coffee (hunk 3 above)
Comment (esoliani) [inline app/scripts/directives/pupilteacherratiochart.coffee:14]: sounds good! I will change it

File: app/scripts/directives/pupilteacherratiochart.coffee (hunk 4 above)
Comment (esoliani) [inline app/scripts/directives/pupilteacherratiochart.coffee]: typo, I'll fix it

File: app/views/pupilteacherratiochart.html (hunk 6 above)
Comment (esoliani) [inline app/views/pupilteacherratiochart.html]: delete it

**Pull request merged**